service/matching/version_rule_helpers.go

```diff
@@ -247,25 +249,38 @@ func CleanupRuleTombstones(versioningData *persistencepb.VersioningData,
 func CommitBuildID(timestamp *hlc.Clock,
 	data *persistencepb.VersioningData,
 	req *workflowservice.UpdateWorkerVersioningRulesRequest_CommitBuildId,
+	hasRecentPoller bool,
 	maxAssignmentRules int) (*persistencepb.VersioningData, error) {
-	data = common.CloneProto(data)
+	data = cloneOrMkData(data)
 	target := req.GetTargetBuildId()
+	if !hasRecentPoller && !req.GetForce() {
+		return nil, serviceerror.NewFailedPrecondition(
+			fmt.Sprintf("no compatible poller seen within the last %s, use force=true to commit anyways",
```

Comment:
do we check this twice?

Reply:
oh yeah, good catch. must have been a weird merge thing

Reply:
deleted the second one